tests/yast2_gui/yast2_software_management.pm

```diff
@@ -9,16 +9,129 @@
 # without any warranty.
 
 # Summary: Test YaST2 module for software management
-# Maintainer: Max Lin <mlin@suse.com>
+# Maintainer: Zaoliang Luo <zluo@suse.de>
 
 use base "y2x11test";
 use strict;
 use testapi;
+use version_utils qw(is_sle is_leap is_tumbleweed);
+
+sub sw_single {
+    # check menu entries at first, then continue with View, Search, RPM Groups
+    # and Installation Summary
+    # File - open Load package List
+    assert_and_click 'sw_file_open';
+    assert_and_click 'sw_file_import';
+    assert_and_click 'package-load-package-list_cancel';
+
+    # Package - All packages - Update
+    assert_and_click 'sw_package';
+    # open menu update if newer version available
+    assert_and_click 'all-in-this-list';
+    send_key 'tab';
+    send_key_until_needlematch 'update-if-newer-version-available', 'down';
+    send_key 'ret';
+    # make sure that package menu got closed after return
+    # otherwise following needle can be matched at wrong place
+    assert_screen [qw(all-in-this-list_still-selected sw_configurations)];
+    if (match_has_tag 'all-in-this-list_still-selected') {
+        mouse_click 'to-close-menu';
+    }
+    wait_still_screen;
```

Comment:
A wait_still_screen with default stilltime is IMHO just wasting time. Can we not wait for something specfic? E.g. a needle?